app/src/main/java/vn/sun/music_68/ui/home/recommend/RecommendFragment.java

```diff
@@ -1,29 +1,119 @@
+    private String mGenreApi;
+    private RecommendContract.Presenter mPresenter;
 
     public RecommendFragment() {
 
     }
 
+    public static RecommendFragment getInstance() {
+        if (sInstance == null) {
+            sInstance = new RecommendFragment();
+        }
+        return sInstance;
+    }
+
     @Override
     protected int getLayoutId() {
         return R.layout.fragment_recommend;
     }
 
     @Override
     protected void initView(View view, Bundle savedInstanceState) {
+        initData();
+        mRecyclerNewRelease = view.findViewById(R.id.recycler_music);
+        mRecyclerGenres = view.findViewById(R.id.recycler_genres);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity(),
+                LinearLayoutManager.HORIZONTAL, false);
+        mRecyclerGenres.setLayoutManager(layoutManager);
+        GenreAdapter genreAdapter = new GenreAdapter(getContext(), this, mGenres);
+        mRecyclerGenres.setAdapter(genreAdapter);
+        LinearLayoutManager gridLayoutManager = new GridLayoutManager(getActivity(), 3,
+                RecyclerView.HORIZONTAL, false);
+        mRecyclerNewRelease.setLayoutManager(gridLayoutManager);
+        newReleaseAdapter = new NewReleaseAdapter(getContext(), mTrackList);
+        mRecyclerNewRelease.setAdapter(newReleaseAdapter);
 
     }
 
-    public static RecommendFragment getInstance() {
-        RecommendFragment recommendFragment = new RecommendFragment();
-        return recommendFragment;
+    private void initData() {
+        mGenres = new ArrayList<>();
+        mGenres.add(new Genres(GenreKey.ALL_AUDIO, GenreName.ALL_AUDIO,
+                R.drawable.genres_all_audio));
+        mGenres.add(new Genres(GenreKey.ALL_MUSIC, GenreName.ALL_MUSIC,
+                R.drawable.genres_all_music));
+        mGenres.add(new Genres(GenreKey.ROCK, GenreName.ROCK,
+                R.drawable.genres_alternativerock));
+        mGenres.add(new Genres(GenreKey.AMBIENT, GenreName.AMBIENT,
+                R.drawable.genres_ambient));
+        mGenres.add(new Genres(GenreKey.CLASSICAL, GenreName.CLASSICAL,
+                R.drawable.genres_classical));
+        mGenres.add(new Genres(GenreKey.COUNTRY, GenreName.COUNTRY,
+                R.drawable.genres_country));
+        initPresenter();
+    }
+
+    private void initPresenter() {
+        TrackRepository repository = TrackRepository.getInstance(TrackRemoteDataSource.getInstance(),
+                TrackLocalDataSource.getInstance(getContext())
+        );
+        mPresenter = new RecommendPresenter(this, repository);
+        mGenreApi = StringUtils.initGenreUrl(GenreKey.ALL_MUSIC, 0);
+        mPresenter.getTrack(mGenreApi);
+    }
+
+    @Override
+    public void onLoadSuccess(List<Track> track) {
+        newReleaseAdapter.addTrack(track);
+    }
+
+    @Override
+    public void onItemClickListener(Genres genres) {
+        Intent intent = GenresActivity.getIntent(getActivity());
```

Comment:
Đoạn này thì phải viết như này chứ nhỉ 
```suggestion
        Intent intent = GenresActivity.getIntent(getActivity(), genres);
```
Xử lý việc putExtra ở trong hàm getIntent

Reply:
đoạn này pull sau a mới làm nên chưa viết đúng